stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarBitmapFactory.kt

```diff
@@ -0,0 +1,135 @@
+package io.getstream.chat.android.ui.avatar
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.LinearGradient
+import android.graphics.Paint
+import android.graphics.Shader
+import android.graphics.Typeface
+import android.graphics.drawable.BitmapDrawable
+import coil.Coil
+import coil.request.ImageRequest
+import com.getstream.sdk.chat.utils.LlcMigrationUtils
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.image
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.avatar.AvatarView.Companion.MAX_AVATAR_SECTIONS
+import io.getstream.chat.android.ui.utils.adjustColorLBrightness
+import io.getstream.chat.android.ui.utils.getIntArray
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal class AvatarBitmapFactory(private val context: Context) {
+    private val gradientBaseColors = context.getIntArray(R.array.avatar_gradient_colors)
+
+    internal suspend fun createUserBitmap(user: User, style: AvatarStyle): Bitmap {
+        return createBitmapFromUrl(user.image)
+            ?: createInitialsBitmap(style, LlcMigrationUtils.getInitials(user))
```

Comment:
It would be great to get rid of LlcMigrationUtils at some point. What do you think about converting `getInitials` to extension function?

Reply:
Good idea @adasiewiczr 
`LlcMigrationUtils` is a class that we should try to avoid, so if we are able to extract those method and refactor them, it would be the best solution

Reply:
Yep, also thought about that

Reply:
Moved to extensions in LLC client